superset/utils/core.py

```diff
@@ -1497,6 +1497,25 @@ def get_since_until(
     if time_range and time_range.startswith("Next") and separator not in time_range:
         time_range = _relative_start + separator + time_range
 
+    if (
+        time_range
+        and time_range.startswith("previous calendar week")
+        and separator not in time_range
+    ):
+        time_range = "DATETRUNC(DATEADD(DATETIME('today'), -1, WEEK), WEEK) : LASTDAY(DATEADD(DATETIME('today'), -1, WEEK), WEEK)"  # pylint: disable=line-too-long
```

Comment:
Wait, these SQL expressions won't work across database engines. I think that we should use python logic to come up with time constants here.

Reply:
I think we need to backtrack some of the work from b592cc7e73 as well here

Reply:
Oh nevermind! it looked like SQL to me at first and dug deeper to find the `pyparsing` lib.

Reply:
If I'm reading the code right, these do not seem to be SQL expressions. They are passed to [`datetime_eval`](https://github.com/airbnb/incubator-superset/blob/865beae3d8644bb5e9df156e5fea80c3ec0202a2/superset/utils/core.py#L1540) which will translate them into actual timestamps before sending to SQL queries.

Reply:
I implemented the parser with `pyparsing`, which allows time_range value to be more flexible